src/datadog/tracer.cpp

```diff
@@ -401,21 +412,23 @@ Expected<Span> Tracer::extract_span(const DictReader& reader,
     //
     // First, though, if the `trace_id_high` tag is already set and has a bogus
     // value or a value inconsistent with the trace ID, tag an error.
-    const std::string hex_high = hex(span_data->trace_id.high);
+    const auto hex_high = hex_padded(span_data->trace_id.high);
     const auto extant = std::find_if(
         trace_tags.begin(), trace_tags.end(), [&](const auto& pair) {
           return pair.first == tags::internal::trace_id_high;
         });
     if (extant == trace_tags.end()) {
       trace_tags.emplace_back(tags::internal::trace_id_high, hex_high);
-    } else if (!parse_uint64(extant->second, 16)) {
-      span_data->tags[tags::internal::propagation_error] =
-          "malformed_tid " + extant->second;
-      extant->second = hex_high;
-    } else if (extant->second != hex_high) {
-      span_data->tags[tags::internal::propagation_error] =
-          "inconsistent_tid " + extant->second;
-      extant->second = hex_high;
+    } else {
```

Comment:
Maybe add a comment here, supporting the comment above, describing what's going on. For example, the appearance of `extant->second = hex_high;` twice might be surprising.